core/src/replay_stage.rs

```diff
@@ -1231,23 +1232,37 @@ impl ReplayStage {
         bank_forks: &RwLock<BankForks>,
         my_pubkey: &Pubkey,
         vote_account: &Pubkey,
+        blockstore: &Blockstore,
     ) -> (ProgressMap, HeaviestSubtreeForkChoice) {
-        let (root_bank, frozen_banks) = {
+        let (root_bank, frozen_banks, duplicate_slot_hashes) = {
             let bank_forks = bank_forks.read().unwrap();
+            let duplicate_slots = blockstore
+                .duplicate_slots_iterator(bank_forks.root_bank().slot())
+                .unwrap();
+            let duplicate_slot_hashes = duplicate_slots
+                .filter_map(|slot| bank_forks.bank_hash(slot).map(|hash| (slot, hash)));
```

Comment:
This can be moved outside of this block scope using just the returned `root_bank`, since it doesn't rely on holding the bank_forks lock

Reply:
You still need the bank forks lock to filter get the hashes for the duplicate slots:
`.filter_map(|slot| bank_forks.bank_hash(slot).map(|hash| (slot, hash)));`

Reply:
ah yeah, I'm dumb, ignore this